src/nccl_ofi_rdma.c

```diff
@@ -6024,7 +6024,7 @@ int nccl_net_ofi_rdma_init(const char *provider_filter,
 		goto error;
 	}
 
-	base_devs = calloc(num_devs, sizeof(nccl_net_ofi_rdma_device_t *));
+	base_devs = calloc(num_devs, sizeof(nccl_net_ofi_device_t *));
```

Comment:
The protocol-specific device structs are cast-able to the base dev by design, and depending on which protocol is inintialzed, the global plugin->devs holds either the sendrecv/rdma device pointers. I can see why the tool is complaining though.

Reply:
Regardless of the casting, these are pointers and we definitely don't expect pointers to these objects to differ in size. This is just meant to silence the warnings.